include/flatbuffers/base.h

```diff
@@ -266,7 +266,7 @@ namespace flatbuffers {
 #ifndef FLATBUFFERS_LOCALE_INDEPENDENT
   // Enable locale independent functions {strtof_l, strtod_l,strtoll_l, strtoull_l}.
   #if ((defined(_MSC_VER) && _MSC_VER >= 1800)            || \
-       (defined(_XOPEN_VERSION) && (_XOPEN_VERSION>=700)) && (!defined(__ANDROID_API__) || (defined(__ANDROID_API__) && (__ANDROID_API__>=21))))
+       (defined(__GLIBC__) && defined(_XOPEN_VERSION) && (_XOPEN_VERSION>=700)) && (!defined(__ANDROID_API__) || (defined(__ANDROID_API__) && (__ANDROID_API__>=21))))
```

Comment:
Does FreeBSD CLIB declare `__GLIBC__` macro?
What about __BIONIC__ that uses _XOPEN_VERSION=700 and other CLIB that doesn't declare nor __GLIBC__ neither other macros (as MUSL does)?

MUSL recommends use `uselocale()` call instead of `_l` methods.
https://man7.org/linux/man-pages/man3/uselocale.3.html
Probably it can be an acceptable solution for non-Windows platforms.
The `uselocale()` can be called only once in the Parser using RAII.

Reply:
Indeed, this PR won't work with clib or Bionic.
I don't master enough flatbuffers to add `uselocale()` but if you can provide a PR, I can test that it builds fine on musl.

Reply:
I'm sorry. I will be unavailable until September even for code review.

The main idea is to add into `utils.cpp` or into `idl_parser.cpp` method like this:
- `flatbuffers::LocaleHandle UseLocale(flatbuffers::LocaleHandle loc=DefaultClassic)` - activate locale and return the previous one.
This method can do nothing if `FLATBUFFERS_LOCALE_INDEPENDENT` is defined and non-zero.
The `flatbuffers::LocaleHandle` type can be committed if `UseLocale` is a cpp-local routine in `idl_parser.cpp`.

The `Parser::DoParse` and `Parser::DoParseJson` should create an RAII-object that will call the `UseLocale` in `ctor` and `dtor`. After this, we can limit `FLATBUFFERS_LOCALE_INDEPENDENT` auto-detection only by MSVC and CMake.

You can try this approach and prepare PR. Or we can start it in September.